Add darkmode 1

LucaAllmaier opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently, the switch to change the darkmode has no functionality 


**What is the new behavior (if this is a feature change)?**
- The darkmode switch sets the darkmode to the localStorage
- The darkmode prop is distributed to the child components


**Other information**:
-

LucaAllmaier committed:
Improve code quality

Automatic spaces have been added.

LucaAllmaier committed:
Enable darkmode Switch

The darkmode switch in the footer has been enabled and some variable names have been changed.

LucaAllmaier committed:
Implement darkmode

The className based on the darkMode prop has been set.

LucaAllmaier committed:
Add darkmode handler

A function to set the darkmode on switch has been added. Additionally, the darkmode is saved to localStorage and read from there on start.

Aichnerc commented on src/App.js:
```diff
@@ -15,13 +15,39 @@ import { ScrollToTop } from "./components/atoms";
+  };
+
+  // Handler for all child components who can set the mode
+  _handler = () => {
```
We no longer use `_` infront of local functions. Please rename to `handler`.

Aichnerc commented on src/App.js:
```diff
@@ -35,11 +61,14 @@ class App extends React.Component {
                 <CookieModal />
               </main>
-              <Footer darkMode={this.state.darkMode} />
+              <Footer darkMode={this.state.darkMode} handler={this._handler} />
```
Change it to `this.handler` to not break the program.

Aichnerc commented on src/App.js:
```diff
@@ -35,11 +61,14 @@ class App extends React.Component {
-                <Routes globalProps={{ ...this.state }} />
+              <main className={this.state.darkMode ? "darkMode" : undefined}>
+                <Routes
+                  mode={this.state.darkMode}
```
The mode is no longer required to be passed on because everything happens within the App.js and Footer.

Aichnerc commented on src/components/molecules/Footer/index.jsx:
```diff
@@ -146,12 +148,10 @@ class Footer extends React.PureComponent {
               <MDBSwitch
                 className="switch-red mb-3"
-                checked={this.props.mode}
+                checked={this.props.darkMode}
```
This is not a stateful switch. Implement a `componentDidMount` where you `this.setState({darkMode: this.props.darkMode ? true : false})` and use the state on the `checked` param.

Aichnerc commented on src/components/organisms/sections/CallToAction/index.jsx:
```diff
@@ -33,7 +33,10 @@ class CallToAction extends React.Component {
-      <section id="calltoaction" className={darkMode ? "dark" : "light"}>
+      <section
+        id="calltoaction"
+        className={this.props.darkMode ? "dark" : "light"}
```
This is no longer required once this component is a child of `<main>`.

Aichnerc commented on src/components/organisms/sections/Services/index.jsx:
```diff
@@ -17,7 +17,7 @@ import data from "./serviceList.js";
   render() {
     return (
-      <section id="services">
+      <section id="services" className={this.props.darkMode ? "dark" : "light"}>
```
This is no longer required once this component is a child of `<main>`.

Aichnerc commented on src/components/organisms/sections/Trusted/index.jsx:
```diff
@@ -38,7 +38,7 @@ const data = [
   render() {
     return (
-      <div id="trusted">
+      <div id="trusted" className={this.props.darkMode ? "dark" : "light"}>
```
This is no longer required once this component is a child of `<main>`.

Aichnerc commented on src/components/pages/HomePage/index.jsx:
```diff
@@ -38,7 +38,7 @@ class HomePage extends React.Component {
           <Hero darkMode={globalProps.darkMode} />
           <Services darkMode={globalProps.darkMode} info={this.state.info} />
-          <Trusted darkmode={globalProps.darkMode} />
+          <Trusted darkMode={globalProps.darkMode} />
```
This is no longer required once this component is a child of `<main>`.

Aichnerc reviewed (changes requested):
Should be okay to merge after those fixes.

LucaAllmaier committed:
Improve code quality

The name of the handler function has been changed. Additionally, the passdown od the mode to Routes has been removed.

LucaAllmaier committed:
Add darkmode to state

The value of the darkmode prop has been added to the state.

LucaAllmaier committed:
Remove darkMode in individual components

The pass on of the darkMode prop to indicidual components has been removed.

Aichnerc reviewed (approved):
Should be okay to merge.

schettn commented on src/App.js:
```diff
@@ -15,13 +15,39 @@ import { ScrollToTop } from "./components/atoms";
 
+// Check if dark or light mode
+function getMode() {
+  let mode = localStorage.getItem("mode");
```
Add new line.
Ref: https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md

schettn commented on src/App.js:
```diff
@@ -15,13 +15,39 @@ import { ScrollToTop } from "./components/atoms";
+    return JSON.parse(mode);
+  } else {
+    // Set default
+    localStorage.setItem("mode", false);
```
Add new line because of logical separation.
Ref: https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md

schettn reviewed (changes requested):
Should be ok after few changes

Aichnerc committed:
Improve code quality

Some new lines have been added due to a request by @schettn.

schettn reviewed (approved):
Great work Mr. Allmaier. 👩‍🎨
